File: src/main/java/entities/gestionSuivi/Objectif.java
package entities.gestionSuivi;

import java.sql.Date;

public class Objectif {
    private int id_objectif;
    private int id_user;
    private float poids_Obj;
    private Date dateD;
    private Date dateF;
    private float poids_Act;
    private float taille ;
    private String alergie;
    private String typeObj;
    private String coachName;
    private String firstName;
    private int coachId ;

    public  Objectif (){

    }
    public Objectif(float poids_Obj,Date dateF,float poids_Act,float taille,String alergie,String typeObj,int CoachId){
        this.setPoids_Obj(poids_Obj);
        this.setDateF(dateF);
        this.setPoids_Act(poids_Act);
        this.setTaille(taille);
        this.setAlergie(alergie);
        this.setTypeObj(typeObj);
        this.setCoachId(CoachId);

    }

    public Objectif(int id_objectif,float poids_Obj,Date dateF,float poids_Act,float taille,String alergie,String typeObj,int CoachId){
        this.setId_objectif(id_objectif);
        this.setPoids_Obj(poids_Obj);
        this.setDateF(dateF);
        this.setPoids_Act(poids_Act);
        this.setTaille(taille);
        this.setAlergie(alergie);
        this.setTypeObj(typeObj);
        this.setCoachId(CoachId);

    }


    public int getId_objectif() {
        return id_objectif;
    }

    public void setId_objectif(int id_objectif) {
        this.id_objectif = id_objectif;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public float getPoids_Obj() {
        return poids_Obj;
    }

    public void setPoids_Obj(float poids_Obj) {
        this.poids_Obj = poids_Obj;
    }

    public Date getDateD() {
        return dateD;
    }

    public void setDateD(Date dateD) {
        this.dateD = dateD;
    }

    public Date getDateF() {
        return dateF;
    }

    public void setDateF(Date dateF) {
        this.dateF = dateF;
    }

    public float getPoids_Act() {
        return poids_Act;
    }

    public void setPoids_Act(float poids_Act) {
        this.poids_Act = poids_Act;
    }

    public float getTaille() {
        return taille;
    }

    public void setTaille(float taille) {
        this.taille = taille;
    }

    public String getAlergie() {
        return alergie;
    }

    public void setAlergie(String alergie) {
        this.alergie = alergie;
    }

    public String getTypeObj() {
        return typeObj;
    }

    public void setTypeObj(String typeObj) {
        this.typeObj = typeObj;
    }

    public String getCoachName() {
        return coachName;
    }

    public void setCoachName(String coachName) {
        this.coachName = coachName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public int getCoachId() {
        return coachId;
    }

    public void setCoachId(int coachId) {
        this.coachId = coachId;
    }
}

File: webapp/src/GymPlus/src/Entity/Objectif.php
<?php
namespace App\Entity;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\ObjectifRepository;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ObjectifRepository::class)]
class Objectif
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idobjectif;
    #[ORM\Column]
    #[Assert\NotBlank(message: "Field is required")]
    #[Assert\Type(type: 'float', message: "Only numeric values are allowed for Weight.")]
    #[Assert\Regex(
        pattern: '/^\d+(\.\d+)?$/',
        message: "Only numeric values are allowed for poidsobj."
    )]
    #[Assert\Range(min: 30, max: 150, minMessage: "Field must be at least 30", maxMessage: "Field must be at most 150")]
    private ?float $poidsobj;
    #[ORM\Column]
    private ?\DateTime $dated;

    #[Assert\NotBlank(message: "Field required")]
    #[ORM\Column(type: "datetime", nullable: true)]
    #[Assert\GreaterThanOrEqual(value: "+1 month", message: "Date field must be greater than or equal to one month from today")]
    private ?\DateTime $datef;

    #[ORM\Column(type: "float", nullable: true)]
    #[Assert\NotBlank(message: "Field required")]
    #[Assert\Type(type: 'float', message: "Only numeric values are allowed for Weight.")]
    #[Assert\Range(min: 30, max: 150, minMessage: "Field must be at least 30", maxMessage: "Field must be at most 150")]
    private ?float $poidsact;

    #[ORM\Column(type: "float", nullable: true)]
    #[Assert\NotBlank(message: "Field required")]
    #[Assert\Type(type: 'float', message: "Only numeric values are allowed for Weight.")]
    #[Assert\Range(min: 140, max: 220, minMessage: "Field must be at least 140 cm", maxMessage: "Field must be at most 220cm")]
    private ?float $taille;

    #[ORM\Column]
    #[Assert\Length(max: 20)]
/**
     * @Assert\Regex(
     *     pattern="/^[a-zA-Z\s', ]*$/",
     *     message="Invalid characters detected."
     * )
     */    private ?string $alergie;

    #[ORM\Column]  
    #[Assert\NotBlank(message: "Select an option")]
     private ?string $typeobj;


    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: "userid", referencedColumnName: "id")]
    private ?User $userid;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: "coachid", referencedColumnName: "id")]
    #[Assert\NotBlank(message: "Select an option")]
    private ?User $coachid;

 

    public function getIdobjectif(): ?int
    {
        return $this->idobjectif;
    }

    public function getPoidsobj(): ?float
    {
        return $this->poidsobj;
    }

    public function setPoidsobj(?float $poidsobj): static
    {
        $this->poidsobj = $poidsobj;

        return $this;
    }

    public function getDated(): ?\DateTime
    {
        return $this->dated;
    }

    public function setDated(?\DateTime $dated): static
    {
        $this->dated = $dated;

        return $this;
    }

    public function getDatef(): ?\DateTime
    {
        return $this->datef;
    }

    public function setDatef(?\DateTime $datef): static
    {
        $this->datef = $datef;

        return $this;
    }

    public function getPoidsact(): ?float
    {
        return $this->poidsact;
    }

    public function setPoidsact(?float $poidsact): static
    {
        $this->poidsact = $poidsact;

        return $this;
    }

    public function getTaille(): ?float
    {
        return $this->taille;
    }

    public function setTaille(?float $taille): static
    {
        $this->taille = $taille;

        return $this;
    }

    public function getAlergie(): ?string
    {
        return $this->alergie;
    }

    public function setAlergie(?string $alergie): static
    {
        $this->alergie = $alergie;

        return $this;
    }
    

    public function getTypeobj(): ?string
    {
        return $this->typeobj;
    }

    public function setTypeobj(?string $typeobj): static
    {
        $this->typeobj = $typeobj;

        return $this;
    }

    public function getCoachid(): ?User
    {
        return $this->coachid;
    }

    public function setCoachid(?User $coachid): static
    {
        $this->coachid = $coachid;

        return $this;
    }

    public function getUserid(): ?User
    {
        return $this->userid;
    }

    public function setUserid(?User $userid): static
    {
        $this->userid = $userid;

        return $this;
    }


}
